File: {{ cookiecutter.repo_name }}/cache/src/main/java/{{ cookiecutter.package_dir }}/cache/room/DatabaseModule.java
package {{ cookiecutter.package_name }}.cache.room;

import android.app.Application;
import androidx.room.Room;

import javax.inject.Singleton;

import dagger.Module;
import dagger.Provides;

@Module
public abstract class DatabaseModule {
    @Provides
    @Singleton
    static AppDatabase providesDatabase(Application app) {
        return Room.databaseBuilder(app, AppDatabase.class, "{{ cookiecutter.repo_name }}-db").build();
    }
}

File: {{ cookiecutter.repo_name }}/cache/src/main/java/{{ cookiecutter.package_dir }}/cache/room/DatabaseModule.kt
package {{ cookiecutter.package_name }}.cache.room;

import android.app.Application
import androidx.room.Room

import javax.inject.Singleton

import dagger.Module
import dagger.Provides

@Module
object DatabaseModule {
    @Provides
    @Singleton
    @JvmStatic
    fun providesDatabase(app: Application): AppDatabase {
        return Room.databaseBuilder(app, AppDatabase::class.java, "{{ cookiecutter.repo_name }}-db").build()
    }
}